File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;
        long long  temp = 0, ref = x;
        while(ref > 0) {
            int last = ref % 10;
            temp = temp * 10 + last;
            ref = ref / 10;
        }
        return temp == x;
        
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) return false;
        int n = x;
        long rev = 0;
        while(n != 0) {
            int last = n % 10;
            rev = rev * 10 + last;
            n = n / 10;
        }
        return rev == x;
    }
}